internal/service/probe.go

```diff
@@ -22,21 +25,11 @@ var once sync.Once
 
 var instance *ProbeService
 
-// Probe is the model required by the service to manipulate
-// the resource.
-type Probe struct {
-	Name   string
-	URL    string
-	Status string
-	Delay  uint
-	finish chan bool
-}
-
 // ProbeService is an implementation
 // of the interface controller.ProbeService
 type ProbeService struct {
 	client *http.Client
-	probes map[string]*Probe
+	probes map[string]*model.Probe
```

Comment:
If we store this at the persistence level, we do not need the map anymore but a reference to the persistence layer. See one of my comment below.

Reply:
The idea is to remove this map from the service, and call the persistence layer everytime service layer needs probe info ?

Reply:
I moved this map to the persistence layer, now the service layer only ask the persistence layer when adding/remove/read probes

Reply:
I was for removing totally the `map`. As we have now a database, the `map` will act more as a `cache`. I want to keep things as simple as possible for the moment and caching may be implemented more riguour later.
To conclude, every time we retrieve probes, we ask the db to return values. (we will still need to keep a reference to launched probe to stop them if client asks)